apps/omg_watcher_info/test/omg_watcher_info/db/repo_test.exs

```diff
@@ -60,4 +89,32 @@ defmodule OMG.WatcherInfo.DB.RepoTest do
       end)
     end
   end
+
+  # Prefer using `ExMachina.build_list/3 which uses `OMG.WatcherInfo.Factory.Transaction`
+  # over this function. This function is built to be fast and simple.
+  defp new_transactions(blknum, count) do
+    Enum.reduce(1..count, [], fn index, acc ->
+      [new_transaction(blknum, index) | acc]
+    end)
+  end
+
+  # Prefer using `ExMachina.build/2 which uses `OMG.WatcherInfo.Factory.Transaction`
+  # over this function. This function is built to be fast and simple.
+  #
+  # `ExMachina.params_for/2` could be used here to make use of `OMG.WatcherInfo.Factory.Transaction`.
+  # But the transaction factory does a lot of extra stuff unnecessary for this test. This stripped
+  # down version is about 15x faster. Also using `ExMachina.params_for/2` here also requires some
```

Comment:
How to get the number 15x?

Some other thoughts that don't block this PR.
1. I agree with "The 2 functions differ in the inputs they expect" so we need to diverge from `ExMachina.build/2`
2. I tend to disagree with having another factory function for speed and simplicity, in exchange with having different ways of building data that we need to be aware of and maintain.

Reply:
Regarding the 15x I just ran it with `ExMachina.params_for/2`, dropping the breaking keys. Logged the test start and end time. Then I ran the test with the `new_transactions/3` function and compared times.

My 'optimization' doesn't change the Big-O of the test, but is perceivably faster...instant test result, vs a 15 second wait.

I agree your other points as well regarding test factory stuff. Let's get to that later.